packages/ts/src/components/treemap/index.ts

```diff
@@ -0,0 +1,307 @@
+    }
+
+    // Render tiles
+    const visibleNodes = descendants.filter(d => d.depth > 0)
+    const tiles = this.tiles
+      .selectAll<SVGGElement, TreemapNode<Datum>>(`g.${s.tileGroup}`)
+      .data(visibleNodes, d => d._id)
+    const tilesEnter = tiles
+      .enter()
+      .append('g')
+      .attr('class', s.tileGroup)
+
+    // Computes the rect border radius for a given tile.
+    // The rx and ry values are the minimum of the tile
+    // border radius and some fraction the width of the tile,
+    // based on the tileBorderRadiusFactor config.
+    // This ensures that the tile border radius is not
+    // larger than the tile size, which makes small tiles
+    // look better.
+    const rx = (d: TreemapNode<Datum>): number =>
+      Math.min(config.tileBorderRadius, (d.x1 - d.x0) * config.tileBorderRadiusFactor)
+
+    // Add clipPath elements
+    tilesEnter
+      .append('clipPath')
+      .attr('id', d => `clip-${d._id}`)
+      .append('rect')
+      .attr('rx', rx)
+      .attr('ry', rx)
+
+    // Tile rectangles
+    tilesEnter
+      .append('rect')
+      .classed(s.tile, true)
+
+      // Make the leaf tiles clickable if a click handler is provided
+      .classed(s.clickableTile, d => config.showTileClickAffordance && !d.children)
+
+      .attr('rx', rx)
+      .attr('ry', rx)
+      // Initialize tile positions so that the initial transition is smooth
+      .attr('x', d => d.x0)
+      .attr('y', d => d.y0)
+      .attr('width', d => d.x1 - d.x0)
+      .attr('height', d => d.y1 - d.y0)
+      .style('fill', d => d._fill ?? getColor(d, config.tileColor))
+      .style('opacity', 0)
+      .style('cursor', config.showTileClickAffordance ? d => !d.children ? 'pointer' : null : null)
+
+    tiles.merge(tilesEnter).select(`rect.${s.tile}`)
+      .style('fill', d => d._fill ?? getColor(d, config.tileColor))
+      .style('opacity', 1)
+      .attr('x', d => d.x0)
+      .attr('y', d => d.y0)
+      .attr('width', d => d.x1 - d.x0)
+      .attr('height', d => d.y1 - d.y0)
+
+    // Update clipPath rects
+    tiles.merge(tilesEnter).select('clipPath rect')
+      .attr('x', d => d.x0)
+      .attr('y', d => d.y0)
+      .attr('width', d => d.x1 - d.x0 - config.tilePadding)
+      .attr('height', d => d.y1 - d.y0)
+
+    // Tile labels
+    tilesEnter
+      .append('text')
+      .attr('class', s.label)
+      .attr('clip-path', d => `url(#clip-${d._id})`)
+    // Update all labels first
+    const mergedTiles = tiles.merge(tilesEnter)
+
+    // Hide labels that don't meet criteria
+    mergedTiles.select(`text.${s.label}`)
+      .style('display', d => {
+        const isAllowedNode = config.labelInternalNodes ? true : !d.children
+        return isAllowedNode && this.isTileLargeEnough(d) ? null : 'none'
+      })
+      // Make the internal labels semibold
+      .style('font-weight', d => d.children ? '500' : 'normal')
+
+    // Update visible labels
+    mergedTiles.select(`text.${s.label}`)
+      .filter(d => {
```

Comment:
Instead of filtering the selection here, I would assign a dedicated class name to the corresponding labels and selected them by it.